src/tqdm_publisher/_demos/_single_bar/_server.py

```diff
@@ -25,39 +25,45 @@ def run_function_on_progress_update(format_dict: dict) -> None:
         on outside parameters must be achieved by defining those fields at an outer scope and defining this
         server-specific callback inside the local scope.
 
-        In this demo, we will execute the `client_callback` whose protocol is known only to the WebSocketHandler.
+        In this demo, we will execute the `progress_callback` whose protocol is known only to the WebSocketHandler.
+
+        This specifically requires the `id` of the progress bar and the `format_dict` of the TQDM instance.
         """
-        client_callback(progress_bar_id=progress_bar_id, format_dict=format_dict)
+        progress_callback(id=progress_bar.id, format_dict=format_dict)
 
     progress_bar.subscribe(callback=run_function_on_progress_update)
 
     for task_duration in progress_bar:
         time.sleep(task_duration)
 
 
-async def handler(websocket: websockets.WebSocketServerProtocol) -> None:
-    """Handle messages from the client and manage the client connections."""
+def send_message_to_client(*, websocket: websockets.WebSocketServerProtocol, message: dict) -> None:
+    """
+    Send a message to a specific client.
 
-    def forward_progress_to_client(*, progress_bar_id: str, format_dict: dict) -> None:
-        """
-        This is the function that will run on every update of the TQDM object.
+    This expects a WebSocket connection and a message (dict) to send.
+    """
 
-        It will forward the progress to the client.
-        """
-        asyncio.run(
-            websocket.send(message=json.dumps(obj=dict(progress_bar_id=progress_bar_id, format_dict=format_dict)))
-        )
+    asyncio.run(websocket.send(message=json.dumps(obj=message)))
+
+
+async def handler(websocket: websockets.WebSocketServerProtocol) -> None:
+    """Handle messages from the client and manage the client connections."""
 
     # Wait for messages from the client
     async for message in websocket:
         message_from_client = json.loads(message)
 
         if message_from_client["command"] == "start":
+
+            # Start the progress bar in a separate thread
             thread = threading.Thread(
                 target=start_progress_bar,
+                # On each update of the progress bar, send this update to the requesting client
                 kwargs=dict(
-                    progress_bar_id=message_from_client["progress_bar_id"],
-                    client_callback=forward_progress_to_client,
+                    progress_callback=lambda id, format_dict: send_message_to_client(
```

Comment:
FYI, in addition to making the keywords more explicit, one big reason I changed everything previously away from `id` is that it is a Python built-in reserved word and using it here overrides the definition - not generally a good idea

Common conventions to avoid this with respect to any built-in are to either (a) append `_` to the end, such as `id_`, or (b) just give it a slightly more specific name (as in, 'what particular ID' or 'and ID for what?')

Reply:
Ah gotcha